pyanaconda/modules/network/nm_client.py

```diff
@@ -498,6 +498,8 @@ def create_connections_from_ksdata(nm_client, network_data, device_name, ifname_
             s390cfg = get_s390_settings(device_name)
             _update_wired_connection_with_s390_settings(con, s390cfg)
 
+    update_connection_ethernet_settings_from_ksdata(con, network_data)
```

Comment:
I don't understand why this is at top level of the function here, while above you decide what kind of connection this is, and one of the options seems to be explicitly this same thing - ethernet:
```
    # type "802-3-ethernet"
    if is_ethernet_device(nm_client, device_name): 
        (...ethernet stuff...)

    update_connection_ethernet_settings_from_ksdata(con, network_data)
```
Maybe I'm mistaking them for each another because there isn't any indication how much ethernet-ish the new function is, except for its name?

Reply:
It applies also to virtual devices, eg bond, but as NM  "802-3-ethernet" setting (it would be applied by NM to ports/slaves for virtual device connections). But your point is very good as for infiniband devices it is "infiniband" setting.
I think I should update the PR.

Reply:
It might be just a question of naming things :)